flow/diff_context.cc

```diff
@@ -163,16 +189,17 @@ void DiffContext::AddLayerBounds(const SkRect& rect) {
   // During painting we cull based on non-overriden transform and then
   // override the transform right before paint. Do the same thing here to get
   // identical paint rect.
-  auto transformed_rect =
-      ApplyFilterBoundsAdjustment(state_.transform.mapRect(rect));
-  if (transformed_rect.intersects(state_.cull_rect)) {
-    auto paint_rect = state_.transform_override
-                          ? ApplyFilterBoundsAdjustment(
-                                state_.transform_override->mapRect(rect))
-                          : transformed_rect;
-    rects_->push_back(paint_rect);
+  auto transformed_rect = ApplyFilterBoundsAdjustment(MapRect(rect));
+  if (transformed_rect.intersects(clip_tracker_.device_cull_rect())) {
```

Comment:
We should perhaps add `clip_tracker_.device_content_culled` to avoid having to dup the Rect with the accessor.